components/layout/use-sidebar.js

```diff
@@ -0,0 +1,33 @@
+import { useContext, useCallback } from 'react'
+
+import LayoutContext from './context'
+
+const useSidebar = () => {
+  const [state, modify] = useContext(LayoutContext)
+  const { sidebarId: id, sidebarExpanded: isOpen } = state
+
+  const toggle = useCallback(
+    (force) => {
+      const value = force == null ? !isOpen : force
+      modify({
+        ...state,
```

Comment:
This is tricky and can lead to bugs. I would rather remove useCallback or remove this line. You would need to put all values from state to second useCallback param.

Here is an example what I mean. If you change sidebar id and then call toggle function the sidebar id gets overidden by old value because of useCallback closure.

Reply:
Thank you for reviewing. I see what you mean, thank you wor pointing it. I will remove `useCallback`